tests/test_utils.py

```diff
@@ -48,6 +49,77 @@ def test_list_exchanges(capsys):
     assert re.search(r"^bittrex$", captured.out, re.MULTILINE)
 
 
+def test_list_timeframes(capsys):
+
+    args = [
+        "list-timeframes",
+    ]
+    pargs = get_args(args)
+    pargs['config'] = None
+    with pytest.raises(OperationalException,
+                       match=r"This command requires a configured exchange.*"):
+        start_list_timeframes(pargs)
+
+    # Test with --config config.json.example
+    args = [
+        '--config', 'config.json.example',
+        "list-timeframes",
+    ]
+    start_list_timeframes(get_args(args))
+    captured = capsys.readouterr()
+    assert re.match("Timeframes available for the exchange `bittrex`: "
+                    "1m, 5m, 30m, 1h, 1d",
+                    captured.out)
+
+    # Test with --exchange bittrex
+    args = [
+        "list-timeframes",
+        "--exchange", "bittrex",
+    ]
+    start_list_timeframes(get_args(args))
+    captured = capsys.readouterr()
+    assert re.match("Timeframes available for the exchange `bittrex`: "
+                    "1m, 5m, 30m, 1h, 1d",
+                    captured.out)
```

Comment:
the way this is tested makes it fail whenever ccxt (or rather the exchange) changes the timeframes they offer.
Now i don't expect this to happen very often - but it could - so maybe using a "loser" regex could be benefitial (allowing more timeframes to be added in the middle / in the end).

The way this is added to ccxt:
https://github.com/ccxt/ccxt/blob/5b6f039a639fc69ae575af8d627e330a7656cbb9/python/ccxt/bittrex.py#L58-L63

Edit: Yes, i had to try the multiline review comment - it's not really tied to these 3 lines.

Reply:
> the way this is tested makes it fail whenever ccxt (or rather the exchange) changes the timeframes they offer.
> Now i don't expect this to happen very often - but it could - so maybe using a "loser" regex could be benefitial (allowing more timeframes to be added in the middle / in the end).

Yes, I guessed it and assumed we'll need to adapt the tests in this case... Should this be completely mocked instead? (And the test shortened to work with one mocked exchange data?)

Reply:
yes (but not shortening) - see the changes i've just pushed.
mocking it to 2 different things has the advantage that you cannot force it "somewhere" to be this output...

Reply:
ah, thanks alot!